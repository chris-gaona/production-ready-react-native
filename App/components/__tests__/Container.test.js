import React from 'react'
import { View } from 'react-native'
import renderer from 'react-test-renderer'

import { Container, styles } from '../Container'

it('should export a styles object', () => {
  expect(typeof styles).toBe('object')
})

it('should render without children', () => {
  const rendered = renderer.create(<Container />).toJSON()
  expect(rendered).toBeTruthy()
})

it('should render children props', () => {
  const rendered = renderer
    .create(<Container>
      <View />
            </Container>)
    .toJSON()
  // toMatchSnapshot creates a snapshot
  expect(rendered).toMatchSnapshot()
})

it('should use the specified backgroundColor if provided', () => {
  const rendered = renderer.create(<Container backgroundColor="red" />).toJSON()
  // toMatchSnapshot creates a snapshot
  expect(rendered).toMatchSnapshot()
})
