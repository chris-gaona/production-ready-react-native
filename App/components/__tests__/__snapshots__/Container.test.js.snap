// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render children props 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#4F6D7A",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View />
</RCTSafeAreaView>
`;

exports[`should use the specified backgroundColor if provided 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#4F6D7A",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
